deafrica_waterbodies/make_polygons.py

```diff
@@ -251,456 +214,271 @@ def get_polygons_from_dataset(
-        using the primary and secondary thresholds.
-
+    gpd.GeoDataFrame
+        waterbody polygons.
     """
-    # Set up the primary and secondary thresholds.
-    minimum_wet_thresholds = [secondary_threshold, primary_threshold]
-
-    # Get the tile id and tile object.
-    tile_id = tile[0]
-    tile_object = tile[1]
+    # Load and threshold the WOfS All Time Summary tile.
+    xr_detection, xr_extent = load_wofs_frequency(
+        tile=tile,
+        grid_workflow=grid_workflow,
+        plugin=plugin,
+        dask_chunks=dask_chunks,
+        min_valid_observations=min_valid_observations,
+        min_wet_thresholds=min_wet_thresholds,
+        land_sea_mask_fp=land_sea_mask_fp,
+    )
 
-    # Generate the waterbody polygons using the primary and secondary thresholds,
-    # from the tile.
+    # Get the crs.
     try:
-        _log.info(f"Generating water body polygons for tile {tile_id}")
-
-        # Load the data for the tile.
-        wofs_alltime_summary = grid_workflow.load(tile_object, dask_chunks=dask_chunks).squeeze()
-
-        # Load the land sea mask.
-        if land_sea_mask_fp:
-            land_sea_mask = rio_slurp_xarray(
-                fname=land_sea_mask_fp,
-                gbox=wofs_alltime_summary.geobox,
-                resampling=resampling_method,
-            )
-
-            # Filter the land sea mask.
-            boolean_land_sea_mask = filter_land_sea_mask(land_sea_mask)
-
-            # Mask the WOfS All-Time Summary dataset using the boolean land sea mask.
-            wofs_alltime_summary = wofs_alltime_summary.where(boolean_land_sea_mask)
-
-        # Set the no-data values to nan.
-        # Masking here is done using the frequency measurement because for multiple
-        # areas NaN values are present in the frequency measurement but the
-        # no data value -999 is not present in the count_clear and
-        # count_wet measurements.
-        # Note: it seems some pixels with NaN values in the frequency measurement
-        # have a value of zero in the count_clear and/or the count_wet measurements.
-        wofs_alltime_summary = wofs_alltime_summary.where(~np.isnan(wofs_alltime_summary.frequency))
-
-        # Mask pixels not observed at least min_valid_observations times.
-        wofs_alltime_summary_valid_clear_count = (
-            wofs_alltime_summary.count_clear >= min_valid_observations
-        )
+        output_crs = xr_detection.geobox.crs
+    except Exception as error:
+        _log.exception(error)
+        output_crs = xr_extent.geobox.crs
 
-        # Generate the polygons.
-        generated_polygons = {}
-        for threshold in minimum_wet_thresholds:
-            # Mask any pixels whose frequency of water detection is less than the threshold.
-            wofs_alltime_summary_valid_wetness = wofs_alltime_summary.frequency > threshold
+    # Convert to numpy arrays for image processing.
+    np_detection = xr_detection.to_numpy().astype(int)
```

Comment:
To save memory, we could use the same change here that we made to the conflux processing -- use xr_detection.values.astype(int) (and same for xr_extent)